src/systemcmds/pwm/pwm.c

```diff
@@ -649,18 +671,26 @@ pwm_main(int argc, char *argv[])
 						}
 					}
 
-					warnx("User abort\n");
+					PX4_INFO("User abort\n");
 					return 0;
 				}
 			}
 
-			usleep(2000);
+			/* Delay longer than the max Oneshot duration */
+
+			usleep(2542);
```

Comment:
No. It is 1 Ms and it will be rounded up.  The  factors that one need to consider is 1) The async nature of the request for the delay, 2) The priority of the task and the interrupt load.

Here is the same code with usleep(1) 

![image](https://cloud.githubusercontent.com/assets/1945821/23823335/41add048-0604-11e7-9a82-da7e54c506bc.png)
 If you zoom in you will see that I am sampling at 100Ms.
Delta A is 1.93 ms, Delta B is 3.6 Ms and Delta C is 1.58